File: TriCoachAI/Models/Enums.swift
import Foundation

enum Sex: String, Codable, CaseIterable, Identifiable {
    case male, female, other
    var id: String { rawValue }

    var label: String {
        switch self {
        case .male: return "Homme"
        case .female: return "Femme"
        case .other: return "Autre"
        }
    }
}

enum AthleteLevel: String, Codable, CaseIterable, Identifiable {
    case beginner, intermediate, advanced
    var id: String { rawValue }

    var label: String {
        switch self {
        case .beginner: return "Débutant"
        case .intermediate: return "Intermédiaire"
        case .advanced: return "Avancé"
        }
    }
}

enum GoalType: String, Codable, CaseIterable, Identifiable {
    case triathlonSprint, triathlonOlympic, duathlon
    case run10k, halfMarathon, marathon
    case ironman, halfIronman
    case improveVMA, weightLoss, generalEndurance

    var id: String { rawValue }

    var label: String {
        switch self {
        case .triathlonSprint: return "Triathlon Sprint"
        case .triathlonOlympic: return "Triathlon Olympique"
        case .duathlon: return "Duathlon"
        case .run10k: return "10 km"
        case .halfMarathon: return "Semi-marathon"
        case .marathon: return "Marathon"
        case .ironman: return "Ironman"
        case .halfIronman: return "Semi-Ironman (70.3)"
        case .improveVMA: return "Améliorer sa VMA"
        case .weightLoss: return "Perdre du poids"
        case .generalEndurance: return "Améliorer l'endurance"
        }
    }

    /// Sports involved, used by the plan engine to balance weekly sessions.
    var sports: [SportType] {
        switch self {
        case .triathlonSprint, .triathlonOlympic, .ironman, .halfIronman:
            return [.swim, .bike, .run, .brick]
        case .duathlon:
            return [.run, .bike, .brick]
        case .run10k, .halfMarathon, .marathon, .improveVMA:
            return [.run]
        case .weightLoss, .generalEndurance:
            return [.run, .bike, .swim]
        }
    }

    /// Approximate total weeks recommended end-to-end (used as a default when
    /// the user picks "durée libre jusqu'à une course cible").
    var recommendedWeeks: Int {
        switch self {
        case .run10k, .improveVMA: return 8
        case .halfMarathon, .duathlon, .triathlonSprint: return 12
        case .marathon, .triathlonOlympic, .halfIronman: return 16
        case .ironman: return 24
        case .weightLoss, .generalEndurance: return 12
        }
    }
}

enum GoalPriority: String, Codable, CaseIterable, Identifiable {
    case a, b, c
    var id: String { rawValue }
    var label: String { "Priorité \(rawValue.uppercased())" }
}

enum GoalStatus: String, Codable {
    case active, achieved, abandoned
}

enum PlanStatus: String, Codable {
    case draft, active, completed, archived
}

enum MacrocyclePhase: String, Codable, CaseIterable {
    case base, build, peak, taper, transition

    var label: String {
        switch self {
        case .base: return "Base"
        case .build: return "Développement"
        case .peak: return "Affûtage spécifique"
        case .taper: return "Affûtage final"
        case .transition: return "Transition"
        }
    }
}

enum SportType: String, Codable, CaseIterable, Identifiable {
    case run, bike, swim, brick, strength, rest
    var id: String { rawValue }

    var label: String {
        switch self {
        case .run: return "Course à pied"
        case .bike: return "Vélo"
        case .swim: return "Natation"
        case .brick: return "Enchaînement (brick)"
        case .strength: return "Renforcement"
        case .rest: return "Repos"
        }
    }

    var systemImage: String {
        switch self {
        case .run: return "figure.run"
        case .bike: return "figure.outdoor.cycle"
        case .swim: return "figure.pool.swim"
        case .brick: return "arrow.triangle.2.circlepath"
        case .strength: return "dumbbell"
        case .rest: return "bed.double"
        }
    }
}

enum WorkoutIntensity: String, Codable, CaseIterable {
    case easy, moderate, hard

    var label: String {
        switch self {
        case .easy: return "Facile"
        case .moderate: return "Modérée"
        case .hard: return "Difficile"
        }
    }
}

enum WorkoutStatus: String, Codable {
    case planned, completed, skipped, modified
}

enum ActivitySource: String, Codable {
    case healthKit, strava, manual
}

enum AdaptationTrigger: String, Codable {
    case missedWorkout, overperformance, underperformance, highFatigue, injuryFlag, lowRecovery, physiologicalStrain

    var label: String {
        switch self {
        case .missedWorkout: return "Séance ratée"
        case .overperformance: return "Surperformance"
        case .underperformance: return "Sous-performance"
        case .highFatigue: return "Fatigue élevée"
        case .injuryFlag: return "Alerte blessure"
        case .lowRecovery: return "Récupération faible"
        case .physiologicalStrain: return "Dérive cardiaque"
        }
    }
}

File: TriCoachAndroid/app/src/main/java/com/tricoach/android/models/Enums.kt
package com.tricoach.android.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class Sex {
    @SerialName("male") MALE,
    @SerialName("female") FEMALE,
    @SerialName("other") OTHER,
}

@Serializable
enum class AthleteLevel {
    @SerialName("beginner") BEGINNER,
    @SerialName("intermediate") INTERMEDIATE,
    @SerialName("advanced") ADVANCED,
}

@Serializable
enum class GoalType {
    @SerialName("triathlonSprint") TRIATHLON_SPRINT,
    @SerialName("triathlonOlympic") TRIATHLON_OLYMPIC,
    @SerialName("duathlon") DUATHLON,
    @SerialName("run10k") RUN_10K,
    @SerialName("halfMarathon") HALF_MARATHON,
    @SerialName("marathon") MARATHON,
    @SerialName("ironman") IRONMAN,
    @SerialName("halfIronman") HALF_IRONMAN,
    @SerialName("improveVMA") IMPROVE_VMA,
    @SerialName("weightLoss") WEIGHT_LOSS,
    @SerialName("generalEndurance") GENERAL_ENDURANCE,
}

@Serializable
enum class GoalPriority {
    @SerialName("a") A,
    @SerialName("b") B,
    @SerialName("c") C,
}

@Serializable
enum class GoalStatus {
    @SerialName("active") ACTIVE,
    @SerialName("achieved") ACHIEVED,
    @SerialName("abandoned") ABANDONED,
}

@Serializable
enum class TimeSlot {
    @SerialName("earlyMorning") EARLY_MORNING,
    @SerialName("morning") MORNING,
    @SerialName("lunch") LUNCH,
    @SerialName("afternoon") AFTERNOON,
    @SerialName("evening") EVENING,
}

@Serializable
enum class WorkoutIntensity {
    @SerialName("easy") EASY,
    @SerialName("moderate") MODERATE,
    @SerialName("hard") HARD,
}

@Serializable
enum class WorkoutStatus {
    @SerialName("planned") PLANNED,
    @SerialName("completed") COMPLETED,
    @SerialName("skipped") SKIPPED,
    @SerialName("modified") MODIFIED,
}

@Serializable
enum class SportType {
    @SerialName("run") RUN,
    @SerialName("bike") BIKE,
    @SerialName("swim") SWIM,
    @SerialName("brick") BRICK,
    @SerialName("strength") STRENGTH,
    @SerialName("rest") REST,
}

@Serializable
enum class AdaptationTrigger {
    @SerialName("missedWorkout") MISSED_WORKOUT,
    @SerialName("overperformance") OVERPERFORMANCE,
    @SerialName("underperformance") UNDERPERFORMANCE,
    @SerialName("highFatigue") HIGH_FATIGUE,
    @SerialName("injuryFlag") INJURY_FLAG,
    @SerialName("lowRecovery") LOW_RECOVERY,
    @SerialName("physiologicalStrain") PHYSIOLOGICAL_STRAIN,
}

@Serializable
enum class MacrocyclePhase {
    @SerialName("base") BASE,
    @SerialName("build") BUILD,
    @SerialName("peak") PEAK,
    @SerialName("taper") TAPER,
    @SerialName("transition") TRANSITION,
}

@Serializable
enum class PlanStatus {
    @SerialName("draft") DRAFT,
    @SerialName("active") ACTIVE,
    @SerialName("completed") COMPLETED,
    @SerialName("archived") ARCHIVED,
}

@Serializable
enum class ActivitySource {
    @SerialName("healthKit") HEALTH_KIT,
    @SerialName("strava") STRAVA,
    @SerialName("manual") MANUAL,
    @SerialName("garmin") GARMIN,
    @SerialName("healthConnect") HEALTH_CONNECT,
}

@Serializable
enum class MealType {
    @SerialName("breakfast") BREAKFAST,
    @SerialName("lunch") LUNCH,
    @SerialName("dinner") DINNER,
    @SerialName("snack") SNACK,
}

@Serializable
enum class DietaryTag {
    @SerialName("vegetarian") VEGETARIAN,
    @SerialName("chickenOnly") CHICKEN_ONLY,
    @SerialName("pescatarian") PESCATARIAN,
    @SerialName("omnivore") OMNIVORE,
}

@Serializable
enum class EffortProfile {
    @SerialName("carbLoad") CARB_LOAD,
    @SerialName("recovery") RECOVERY,
    @SerialName("light") LIGHT,
    @SerialName("balanced") BALANCED,
}

@Serializable
enum class RecipeCategory {
    @SerialName("dips") DIPS,
    @SerialName("cookies") COOKIES,
    @SerialName("ovenBaked") OVEN_BAKED,
    @SerialName("stew") STEW,
    @SerialName("sandwich") SANDWICH,
    @SerialName("dessert") DESSERT,
    @SerialName("toast") TOAST,
    @SerialName("salad") SALAD,
    @SerialName("pie") PIE,
    @SerialName("vegetarian") VEGETARIAN,
    @SerialName("cake") CAKE,
    @SerialName("pasta") PASTA,
    @SerialName("soup") SOUP,
}

@Serializable
enum class PrepTimeBucket {
    @SerialName("under15") UNDER_15,
    @SerialName("min15to30") MIN_15_TO_30,
    @SerialName("min30to45") MIN_30_TO_45,
    @SerialName("min45to60") MIN_45_TO_60,
    @SerialName("over60") OVER_60,
}

@Serializable
enum class GroceryAisle {
    @SerialName("butcher") BUTCHER,
    @SerialName("bakery") BAKERY,
    @SerialName("grocery") GROCERY,
    @SerialName("produce") PRODUCE,
    @SerialName("fishmonger") FISHMONGER,
    @SerialName("fresh") FRESH,
    @SerialName("frozen") FROZEN,
}

@Serializable
enum class MenuSelectionStatus {
    @SerialName("proposed") PROPOSED,
    @SerialName("confirmed") CONFIRMED,
}
